ga4gh/backend.py

```diff
@@ -531,6 +539,96 @@ def callSetsGenerator(self, request):
                 return self._noObjectGenerator()
             return self._singleObjectGenerator(callSet)
 
+    def rnaQuantificationGenerator(self, request):
+        """
+        Returns a generator over the (rnaQuantification, nextPageToken) pairs
+        defined by the specified request.
+        """
+        if request.rnaQuantificationId is None:
+            #TODO; remove hard coded dataset ID
+            dataset = self.getDataset('MWtnLXAzLXN1YnNldA==')
+            return self._objectListGenerator(request,
+                                             dataset.getRnaQuantifications())
+        else:
+            compoundId = datamodel.RnaQuantificationCompoundId.parse(request.rnaQuantificationId)
+            rnaQuantificationName = compoundId.rnaQuantification
+            dataset = self.getDataset(compoundId.datasetId)
+            try:
+                rnaQuant = dataset.getRnaQuantification(rnaQuantificationName)
+            except exceptions.RnaQuantificationNotFoundException:
+                return self._noObjectGenerator()
+            return self._singleObjectGenerator(rnaQuant)
+
+    def expressionLevelGenerator(self, request):
+        expressionLevelId = request.expressionLevelId
+        featureGroupId = request.featureGroupId
+        rnaQuantificationId = request.rnaQuantificationId
+        currentIndex = 0
+        #TODO; remove hard coded dataset ID
+        dataset = self.getDataset('MWtnLXAzLXN1YnNldA==')
+        if request.pageToken is not None:
+            currentIndex, = _parsePageToken(request.pageToken, 1)
+        if rnaQuantificationId is not None:
+            rnaQuantificationIds = [rnaQuantificationId]
+        else:
+            rnaQuantificationIds = dataset.getRnaQuantificationIds()
+        rnaQuantifications = dataset.getRnaQuantificationIdMap()
+        for rnaQuantId in rnaQuantificationIds:
+            rnaQuant = rnaQuantifications[rnaQuantId]
+            expressionLevelIterator = rnaQuant.getExpressionLevel(
+                expressionLevelId, featureGroupId)
+            expressionLevelData = next(expressionLevelIterator, None)
+            while (expressionLevelData is not None and
+                    currentIndex < self._defaultPageSize):
+                nextExpressionLevelData = next(expressionLevelIterator, None)
+                nextPageToken = None
+                if nextExpressionLevelData is not None:
```

Comment:
is it not possible to eliminate some of this logic with some of our already-written utility functions in `backend`?  Like `_topLevelObjectGenerator` or a similar method?